pkg/getter/ocigetter.go

```diff
@@ -49,6 +49,28 @@ func (g *OCIGetter) get(href string) (*bytes.Buffer, error) {
 		return nil, err
 	}
 
+	// In case any TLS setting is set, recreate the registry client
+	opts := []registry.ClientOption{}
```

Comment:
this change mostly looks good and thanks for adding lots of tests! My main concern is the registryclient being recreated in several places.. is there some way we create it just once in the CLI configuration?

Reply:
Thanks for the comments. I added another commit to move the client creation into `PersistentPreRunE` to create the registry client just before command run with the tls options. However, I am not able to find a way to make `helm install/upgrade/...` to use the same client without modifying existing function signature. Currently, `helm pull` uses the global registry client, but `helm install/upgrade/...` use registry client created in OCIGetter and there is no way to pass the global registry client to NewOCIGetter without modifying `LocateChart`. With this, there is a bug about registry config file. If user specifies registry config file with global ` --registry-config`, `helm install/upgrade/...` won't honor it.